From: Hideyuki Yamashita <yamashita.hideyuki@ntt-tx.co.jp>
Date: Wed, 15 Jan 2020 16:41:01 +0900
Subject: [PATCH] spp-ctl: fix exception when receiving large data

Upon receiving large size response, spp-ctl tries
to refer self variable which does not exist
and thus exception takes place.
This patch tries to refer Class Object instead.

Fixes: f747dcf8 ("spp-ctl: add updating for no existing sec procs")

Signed-off-by: Hideyuki Yamashita <yamashita.hideyuki@ntt-tx.co.jp>
Signed-off-by: Yasufumi Ogawa <yasufum.o@gmail.com>
---
 src/spp-ctl/spp_ctl.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/spp-ctl/spp_ctl.py b/src/spp-ctl/spp_ctl.py
index ea19aa6..9e11097 100644
--- a/src/spp-ctl/spp_ctl.py
+++ b/src/spp-ctl/spp_ctl.py
@@ -124,7 +124,7 @@ class Controller(object):
             data = conn.recv(MSG_SIZE)
             if data and len(data) == MSG_SIZE:
                 # could not receive data at once. recieve remining data.
-                data += self._continue_recv(conn)
+                data += Controller._continue_recv(conn)
             if data:
                 data = data.decode()
         except Exception as e:
-- 
2.17.1

